lib/src/generate/build_impl.dart

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../asset/asset.dart';
+import '../asset/exceptions.dart';
+import '../asset/id.dart';
+import '../asset/reader.dart';
+import '../asset/writer.dart';
+import '../asset_graph/graph.dart';
+import '../asset_graph/node.dart';
+import '../builder/builder.dart';
+import '../builder/build_step_impl.dart';
+import '../package_graph/package_graph.dart';
+import 'build_result.dart';
+import 'exceptions.dart';
+import 'input_set.dart';
+import 'phase.dart';
+
+/// Class which manages running builds.
+class BuildImpl {
+  final AssetGraph _assetGraph;
+  final AssetReader _reader;
+  final AssetWriter _writer;
+  final PackageGraph _packageGraph;
+  final List<List<Phase>> _phaseGroups;
+  Map<String, Set<AssetId>> _inputsByPackage;
```

Comment:
nit: Make this `final` & assign right away.

Change the assignment in `_initializeInputsByPackage` to a call to `clear()`